web/blog/zeek-in-the-shell/index.md

```diff
@@ -0,0 +1,179 @@
+JSON](https://suricata.readthedocs.io/en/latest/output/eve/eve-json-output.html)
+by default, which is equivalent to Zeek's streaming JSON output, with the
+difference being that Zeek's `_path` field is called `event_type` in Suricata
+logs.
+
+```bash title=suricatify
+#!/bin/sh
+suricata -r /dev/stdin \
+  --set outputs.1.eve-log.filename=/dev/stdout \
+  --set logging.outputs.0.console.enabled=no
+```
+
+Let's break this down:
+
+- The `--set` option take a `name=value` parameter that overrides the settings
+  in your `suricata.yaml` config file.
+- The key `outputs.1.eve-log.filename` refers to the `outputs` array, takes
+  element at index `1`, treats that as object and goes to the nested field
+  `eve-log.filename`. Setting `/dev/stdout` as filename makes Suricata write to
+  stdout.
+- We must set `logging.outputs.0.console.enabled` to `no` because Suricata
+  doesn't write startup log messages on stderr. Since they are no valid JSON, we
+  would otherwise create an invalid JSON output stream.
+
+## User-defined Operators
+
+Now that we have both Zeek and Suricata at our fingertips, how can we work with
+their output more easily? This is where Tenzir comes into play-easy
+[pipelines](/docs/language/pipelines) for security teams to acquire,
+[reshape](/docs/user-guides/get-started), and route event data.
+
+For our use case we look at the [`shell`](/docs/operators/shell) operator in
+more detail: it lifts any Unix command line into a pipeline. Here are two
+examples that rely on our two shell scripts:
+
+```bash
+# Zeek
+zcat pcap.gz | zeekify | tenzir \
+  'read zeek-json
+   | where #type == "zeek.conn"
+   | summarize n=count_distinct(id.orig_h) by id.resp_h
+   | sort n desc'
+# Suricata
+zcat pcap.gz | suricatify | tenzir \
+  'read suricata
+   | where #type == "suricata.flow"
+   | summarize n=count_distinct(src_ip) by dst_ip
+   | sort n desc'
+```
+
+The two aggregations actually do the same thing: count the number of unique
+source IP addresses per destination IP address.
+
+It's a bit unwiedly to write such a command line that requires an external shell
+script to work. This where [user-defined operators](/operators/user-defined)
+come into play. You can write a custom `zeek` and `suricata` operator and ditch
+the shell script:
+
+```yaml title="tenzir.yaml"
+tenzir:
+  operators:
+    zeek: >
+     shell "zeek -r - LogAscii::output_to_stdout=T \
+            JSONStreaming::disable_default_logs=T \
+            JSONStreaming::enable_log_rotation=F \
+            json-streaming-logs"
+     | parse zeek-json
+    suricata: >
+     shell "suricata -r /dev/stdin \
+            --set outputs.1.eve-log.filename=/dev/stdout \
+            --set logging.outputs.0.console.enabled=no"
+     | parse suricata
+```
+
+The difference stands out when you look now at the pipeline definition:
+
+```text title=Zeek
+zeek
+| where #type == "zeek.conn"
+| summarize n=count_distinct(id.orig_h) by id.resp_h
+| sort n desc
+```
```

Comment:
Also, we have an open issue for when shell is used as a source. As per the comment above, you maybe do not want do this here, and the issue may not apply to this situation, but just to let you know: https://github.com/tenzir/issues/issues/619

Reply:
Yes, you can resolve this one here.